### ISSUE TITLE
Make D.O.B in Name Database optional.

### ISSUE THREAD

**Issue by Flimsyer** [Linesmerrill/police-cad#153]
Hello Lines Police,

When I was using your CAD, I wanted to put a name into the 'Name Database' however the D.O.B was required before I could search the name. I didn't have the D.O.B information making this a problem for me.

**Describe the solution you'd like**
I would like it if the D.O.B requirement in the 'Name Database' was made optional as I might need to search a name but don't have the D.O.B.

**Implementation Details**
Basically just make it optional and not required so that you can search names even without filling in the D.O.B section

**Possible Side Effects**
I can't see any negative side effects this would have.

Thanks.

**Comment by Linesmerrill** [comment]
Hi @Flimsyer thank you for the suggestion. So we have one option you can take to not have to enter the DOB every time:

### Option 1: Join a community

We removed this required DOB field specifically in communities. So as long as you are in a community, you will no longer need to enter the DOB when searching for a person by name.

### Explanation behind keeping the field if not in a community:

We found that there were a lot of complaints from users about having to filter through 50 or sometimes 100 duplicates of the same name. When users all shared the same database, they could literally create an infinite amount of users with the exact same name. Yes some people abused this and ruined it for everyone else. This then hindered the usefulness of trying to quickly look someone up by only their name. 

So we introduced the DOB field as an additional query to our database to help filter the results returned. This works great to not only improve performance but also increase the user experience due to limited results returned. We understand that this may be inconvient, which is why we took this out if you join a community, but it is necessary in order for us to provide a performant and relaiable service to all of you.

**Comment by Flimsyer** [comment]
Thank you for this. However I’m relatively new to the cad and haven’t explored it all that much. How would I go about joining a community. Thanks Sent from Mail for Windows 10, Ben Crew From: Merrill LinesSent: Wednesday, 8 April 2020 3:31 AMTo: Linesmerrill/police-cadCc: Flimsyer; MentionSubject: Re: [Linesmerrill/police-cad] Make D.O.B in Name Database optional. (#153) Hi @Flimsyer thank you for the suggestion. So we have one option you can take to not have to enter the DOB every time:Option 1: Join a communityWe removed this required DOB field specifically in communities. So as long as you are in a community, you will no longer need to enter the DOB when searching for a person by name.Explanation behind keeping the field if not in a community:We found that there were a lot of complaints from users about having to filter through 50 or sometimes 100 duplicates of the same name. When users all shared the same database, they could literally create an infinite amount of users with the exact same name. Yes some people abused this and ruined it for everyone else. This then hindered the usefulness of trying to quickly look someone up by only their name.So we introduced the DOB field as an additional query to our database to help filter the results returned. This works great to not only improve performance but also increase the user experience due to limited results returned. We understand that this may be inconvient, which is why we took this out if you join a community, but it is necessary in order for us to provide a performant and relaiable service to all of you.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by Linesmerrill** [comment]
@Flimsyer to join a community simply follow these steps:

To join or create a community for free, log in as a Civilian, Police or Fire/EMS and click on the communities button under the top right menu, or at the top right of your screen.

Or just click here: [Communities](https://www.linespolice-cad.com/civ-dashboard#communitiesModal)
